internal/resource-request-operator/broadcaster.go

```diff
@@ -101,58 +120,94 @@ func (b *Broadcaster) onUpdate(oldObj, newObj interface{}) {
 		(newNode.Status.Phase == corev1.NodeTerminated || newNode.Status.Phase == corev1.NodePending) {
 		subResources(currentResources, oldNodeResources)
 	}
-	if err := b.writeResources(currentResources); err != nil {
-		klog.Errorf("OnUpdate error: unable to write allocatable of Node %s: %s", newNode.Name, err)
-	}
+	b.writeClusterResources(currentResources)
 }
 
 // react to a Node Delete.
-func (b *Broadcaster) onDelete(obj interface{}) {
+func (b *Broadcaster) onNodeDelete(obj interface{}) {
 	node := obj.(*corev1.Node)
 	toDelete := &node.Status.Allocatable
-	currentResources := b.allocatable.DeepCopy()
+	currentResources := b.readClusterResources()
 	if node.Status.Phase == corev1.NodeRunning {
 		subResources(currentResources, *toDelete)
-		if err := b.writeResources(currentResources); err != nil {
-			klog.Errorf("OnAdd error: unable to write allocatable of Node %s: %s", node.Name, err)
-		}
+		b.writeClusterResources(currentResources)
 	}
 }
 
-// write cluster resources in thread safe mode.
-func (b *Broadcaster) writeResources(newResources corev1.ResourceList) error {
-	b.offerMutex.Lock()
-	defer b.offerMutex.Unlock()
-	if newResources != nil {
-		b.allocatable = newResources.DeepCopy()
-		return nil
+func (b *Broadcaster) onPodAdd(obj interface{}) {
+	pod := obj.(*corev1.Pod)
+	klog.V(4).Infof("OnPodAdd: Add for pod %s\n", pod.Name)
+	podResources := extractPodResources(pod)
+	currentResources := b.readClusterResources()
+	if clusterID := pod.Labels[forge.LiqoOriginClusterID]; clusterID != "" {
+		klog.V(4).Infof("OnPodAdd: Pod %s passed ClusterID check\n", pod.Name)
+		currentPodsResources := b.readPodResources(clusterID)
+		addResources(currentPodsResources, podResources)
+		b.writePodsResources(clusterID, currentPodsResources)
 	}
+	subResources(currentResources, podResources)
+	b.writeClusterResources(currentResources)
```

Comment:
Would move this above the if check. As this logic is not intuitive at first sight, I feel it would be better to add a couple of comments about why you are subtracting the resources from an accumulator and, if the cluster ID matches, adding them to another one.